### DISCUSSION TITLE
Is there a way to block the deployment to the Vercel if the tests in github actions fail?

### DISCUSSION THREAD

**Discussion by volodymyrmuryn** [vercel/community#2294] [Help] [answered]
### Summary

Hi there, I'm using Vercel and want to implement the logic of blocking the deployments when the unit tests fail. Is there a preferable way of doing this? I'm using the Github Integration and vercel-deployment is not called from the GH actions directly.

Is there a recommended gh-action that I can use directly in the actions to deploy the app? Any information would be helpful.

**Comment by amyegan** [comment, marked as answer]
Hi @volodymyrmuryn! You can create a custom workflow using Vercel CLI to deploy only if tests pass. The following examples don't include testing conditions, but they do provide a good base that you can start from.
- [How can I use the Vercel CLI for custom workflows?](https://vercel.com/guides/using-vercel-cli-for-custom-workflows)
- [How can I use GitHub Actions with Vercel?](https://vercel.com/guides/how-can-i-use-github-actions-with-vercel)
- [GitHub Actions deployment example repo](https://github.com/vercel/examples/tree/main/ci-cd/github-actions)
